src/Appwrite/Platform/Workers/Messaging.php

```diff
@@ -195,14 +195,6 @@ private function processMessage(Database $dbForProject, Document $message): void
                                 if ($detail['status'] === 'failure') {
                                     $deliveryErrors[] = `Failed sending to target {$detail['recipient']} with error: {$detail['error']}`;
                                 }
-
-                                // Deleting push targets when token has expired.
-                                if ($detail['error'] === 'Expired device token.') {
-                                    $target = $dbForProject->findOne('targets', [Query::equal('identifier', [$detail['recipient']])]);
-                                    if ($target instanceof Document && !$target->isEmpty()) {
-                                        $dbForProject->deleteDocument('targets', $target->getId());
-                                    }
-                                }
```

Comment:
We should automatically delete a target if the token is expired/invalid. Imagine if I release my app, 1 million users install + sign up, and are added to the "all users" topic. The next day all but 1 uninstall the app. Now, sending a message to the "all users" topic will _always_ have 999,999 failures.

Reply:
@eldadfux, thoughts?

Reply:
Can we let the maintenance and delete workers collaborate on this? we do similar things in other areas.

Reply:
Another option is to skip them and still count them for usage, this way there's a bigger incentive for the devs to handle the list properly.

Reply:
We could do this in the deletes worker if we mark them as invalid here and then trigger the deletes worker. I don't think we can keep & skip them, since for mobile apps, there's no uninstall hook; developers have no way to do the removal themselves

Reply:
I think it make sense to mark them here or maybe even count consecutive failures.